mlir/lib/Conversion/ArmSMEToLLVM/ArmSMEToLLVM.cpp

```diff
@@ -574,10 +573,17 @@ void mlir::configureArmSMEToLLVMConversionLegality(ConversionTarget &target) {
       arm_sme::aarch64_sme_mopa>();
   target.addLegalDialect<arith::ArithDialect>();
   target.addLegalOp<UnrealizedConversionCastOp>();
+  typeConverter.addConversion([&](VectorType type) -> std::optional<Type> {
+    // There's no LLVM type for SME tiles, but after lowering to intrinsics all
+    // SME vector types should be eliminated.
+    if (arm_sme::isValidSMETileVectorType(type))
+      return type;
+    return std::nullopt;
+  });
```

Comment:
nit: move to `populateArmSMEToLLVMTypeConversion`?

Reply:
I meant a separate populate method that's called from configure (based on what I saw in SPIR-V which looked clean), but it's not so important :)

Reply:
Oh, I was looking at OpenMPToLLVM, which does it like this :)

Reply:
no worries, that's fine :)